src/main/java/com/imsweb/x12/reader/X12Reader.java

```diff
@@ -170,7 +191,9 @@ private void parse(FileType type, Reader reader) throws IOException {
 
         // set up delimiters
         Separators separators = getSeparators(reader);
-        if (separators != null) {
+        boolean consistentVersions = checkVersionsAreConsistent(type, separators, reader);
+
+        if (separators != null && consistentVersions) {
```

Comment:
Why assign `consistentVersions` to a local variable?  Just do the check in the if statement.